libsodium-sys/libsodium

```diff
@@ -0,0 +1 @@
+Subproject commit f01d7c128214e8f976889c8093a42d9cae1abe4b
```

Comment:
Yes, I was in two minds about this.

It would be good to have Windows and non-Windows builds using the exact same build of libsodium, and using the tagged commit would achieve that.

However, I was really keen to use the threadsafe version of `sodium_init()` which will be included in 1.0.11 since this still isn't handled by sodiumoxide, so I selfishly just used tip-of-tree of the stable branch of libsodium.

Ultimately I guess this will be a decision for @dnaq since he'll be the one updating the submodule commit and the tag version for Windows builds on a regular basis if this PR is accepted.  If the general preference is to use the tagged commit, we could probably reduce the maintenance overhead by defining the tag version outside of the platform-specific functions, then for non-Windows builds, do something like a `git fetch` then `git checkout $VERSION`.  That way, only the `VERSION` variable would need to be updated as and when a new release of libsodium lands.